### PR TITLE
Fix compilation on iPad Pro 1st Gen with iPadOS13 and iOS 12

### LINKED ISSUE #124: Swift 5.1: iPad compile errors: ARMV7_NEON, ARMV8_SHA256
swift-nio-ssl commit hash: 7eb03213377903034df1438558a96524837339cf

Context:
Compiling 

# Compiling for iPhoneXR Simulator + iOS12 Device: 

1. Open swift-nio-ssl as a Swift Package in Xcode.
2. Compile for iPhone XR.
3. Observe compile error in test:

      swift-nio-ssl/Tests/NIOSSLTests/NIOSSLIntegrationTest.swift:24:14: Class 'Process' does not exist in module 'Foundation'

# Compiling for: iPadOS 13 device, iOS 12 iPad Pro device (both 1st gen)

1. Open swift-nio-ssl as a Swift Package in Xcode.
1. Compile for the above.
3. Observing compile error:

        CompileC /Volumes/DerivedData-ramdisk/DerivedData/swift-nio-ssl-hezwwgdphjgsdadtxckbgbxhjldw/Build/Intermediates.noindex/swift-nio-ssl.build/Debug-iphoneos/CNIOBoringSSL.build/Objects-normal/arm64/crypto.o /Users/fabiansturm/Dropbox/projects/swift5.1/swift-nio-ssl/Sources/CNIOBoringSSL/crypto/crypto.c normal arm64 c com.apple.compilers.llvm.clang.1_0.compiler (in target 'CNIOBoringSSL' from project 'swift-nio-ssl')
            cd /Users/fabiansturm/Dropbox/projects/swift5.1/swift-nio-ssl
            export LANG=en_US.US-ASCII
            /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x c -target arm64-apple-ios8.0 -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -fmodules -gmodules -fmodules-cache-path=/Volumes/DerivedData-ramdisk/DerivedData/ModuleCache.noindex -fmodules-prune-interval=86400 -fmodules-prune-after=345600 -fbuild-session-file=/Volumes/DerivedData-ramdisk/DerivedData/ModuleCache.noindex/Session.modulevalidation -fmodules-validate-once-per-build-session -Wnon-modular-include-in-framework-module -Werror=non-modular-include-in-framework-module -fmodule-name=CNIOBoringSSL -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wshorten-64-to-32 -Wpointer-sign -Wno-newline-eof -DDEBUG=1 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS13.0.sdk -fstrict-aliasing -Wdeprecated-declarations -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-semicolon-before-method-body -fembed-bitcode-marker -index-store-path /Volumes/DerivedData-ramdisk/DerivedData/swift-nio-ssl-hezwwgdphjgsdadtxckbgbxhjldw/Index/DataStore -I/Volumes/DerivedData-ramdisk/DerivedData/swift-nio-ssl-hezwwgdphjgsdadtxckbgbxhjldw/Build/Products/Debug-iphoneos/include -I/Users/fabiansturm/Dropbox/projects/swift5.1/swift-nio-ssl/Sources/CNIOBoringSSL/include -I/Volumes/DerivedData-ramdisk/DerivedData/swift-nio-ssl-hezwwgdphjgsdadtxckbgbxhjldw/Build/Intermediates.noindex/swift-nio-ssl.build/Debug-iphoneos/CNIOBoringSSL.build/DerivedSources-normal/arm64 -I/Volumes/DerivedData-ramdisk/DerivedData/swift-nio-ssl-hezwwgdphjgsdadtxckbgbxhjldw/Build/Intermediates.noindex/swift-nio-ssl.build/Debug-iphoneos/CNIOBoringSSL.build/DerivedSources/arm64 -I/Volumes/DerivedData-ramdisk/DerivedData/swift-nio-ssl-hezwwgdphjgsdadtxckbgbxhjldw/Build/Intermediates.noindex/swift-nio-ssl.build/Debug-iphoneos/CNIOBoringSSL.build/DerivedSources -F/Volumes/DerivedData-ramdisk/DerivedData/swift-nio-ssl-hezwwgdphjgsdadtxckbgbxhjldw/Build/Products/Debug-iphoneos -F/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/Frameworks -MMD -MT dependencies -MF /Volumes/DerivedData-ramdisk/DerivedData/swift-nio-ssl-hezwwgdphjgsdadtxckbgbxhjldw/Build/Intermediates.noindex/swift-nio-ssl.build/Debug-iphoneos/CNIOBoringSSL.build/Objects-normal/arm64/crypto.d --serialize-diagnostics /Volumes/DerivedData-ramdisk/DerivedData/swift-nio-ssl-hezwwgdphjgsdadtxckbgbxhjldw/Build/Intermediates.noindex/swift-nio-ssl.build/Debug-iphoneos/CNIOBoringSSL.build/Objects-normal/arm64/crypto.dia -c /Users/fabiansturm/Dropbox/projects/swift5.1/swift-nio-ssl/Sources/CNIOBoringSSL/crypto/crypto.c -o /Volumes/DerivedData-ramdisk/DerivedData/swift-nio-ssl-hezwwgdphjgsdadtxckbgbxhjldw/Build/Intermediates.noindex/swift-nio-ssl.build/Debug-iphoneos/CNIOBoringSSL.build/Objects-normal/arm64/crypto.o

        /Users/fabiansturm/Dropbox/projects/swift5.1/swift-nio-ssl/Sources/CNIOBoringSSL/crypto/crypto.c:92:5: error: use of undeclared identifier 'ARMV7_NEON'
            ARMV7_NEON |
            ^
        /Users/fabiansturm/Dropbox/projects/swift5.1/swift-nio-ssl/Sources/CNIOBoringSSL/crypto/crypto.c:95:5: error: use of undeclared identifier 'ARMV8_AES'
            ARMV8_AES |
            ^
        /Users/fabiansturm/Dropbox/projects/swift5.1/swift-nio-ssl/Sources/CNIOBoringSSL/crypto/crypto.c:98:5: error: use of undeclared identifier 'ARMV8_SHA1'
            ARMV8_SHA1 |
            ^
        /Users/fabiansturm/Dropbox/projects/swift5.1/swift-nio-ssl/Sources/CNIOBoringSSL/crypto/crypto.c:101:5: error: use of undeclared identifier 'ARMV8_SHA256'
            ARMV8_SHA256 |
            ^
        /Users/fabiansturm/Dropbox/projects/swift5.1/swift-nio-ssl/Sources/CNIOBoringSSL/crypto/crypto.c:104:5: error: use of undeclared identifier 'ARMV8_PMULL'
            ARMV8_PMULL |
            ^
        5 errors generated.

# Compiling for: iPad OS 13 device, iOS 12 iPad Pro device (both 1st gen)

1. Open swift-nio-ssl as a Swift Package in Xcode.
1. Compile for the above.
2. Observe errors from iPad Pro + additional error below:

        CompileC /Volumes/DerivedData-ramdisk/DerivedData/swift-nio-ssl-hezwwgdphjgsdadtxckbgbxhjldw/Build/Intermediates.noindex/swift-nio-ssl.build/Debug-iphoneos/CNIOBoringSSL.build/Objects-normal/arm64/sha256-armv8.ios.aarch64.o /Users/fabiansturm/Dropbox/projects/swift5.1/swift-nio-ssl/Sources/CNIOBoringSSL/crypto/fipsmodule/sha256-armv8.ios.aarch64.S normal arm64 assembler-with-cpp com.apple.compilers.llvm.clang.1_0.compiler (in target 'CNIOBoringSSL' from project 'swift-nio-ssl')
            cd /Users/fabiansturm/Dropbox/projects/swift5.1/swift-nio-ssl
            export LANG=en_US.US-ASCII
            /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x assembler-with-cpp -target arm64-apple-ios8.0 -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -fmodules -gmodules -fmodules-cache-path=/Volumes/DerivedData-ramdisk/DerivedData/ModuleCache.noindex -fmodules-prune-interval=86400 -fmodules-prune-after=345600 -fbuild-session-file=/Volumes/DerivedData-ramdisk/DerivedData/ModuleCache.noindex/Session.modulevalidation -fmodules-validate-once-per-build-session -Wnon-modular-include-in-framework-module -Werror=non-modular-include-in-framework-module -fmodule-name=CNIOBoringSSL -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wshorten-64-to-32 -Wno-newline-eof -DDEBUG=1 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS13.0.sdk -fstrict-aliasing -Wdeprecated-declarations -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-semicolon-before-method-body -fembed-bitcode-marker -index-store-path /Volumes/DerivedData-ramdisk/DerivedData/swift-nio-ssl-hezwwgdphjgsdadtxckbgbxhjldw/Index/DataStore -I/Volumes/DerivedData-ramdisk/DerivedData/swift-nio-ssl-hezwwgdphjgsdadtxckbgbxhjldw/Build/Products/Debug-iphoneos/include -I/Users/fabiansturm/Dropbox/projects/swift5.1/swift-nio-ssl/Sources/CNIOBoringSSL/include -I/Volumes/DerivedData-ramdisk/DerivedData/swift-nio-ssl-hezwwgdphjgsdadtxckbgbxhjldw/Build/Intermediates.noindex/swift-nio-ssl.build/Debug-iphoneos/CNIOBoringSSL.build/DerivedSources-normal/arm64 -I/Volumes/DerivedData-ramdisk/DerivedData/swift-nio-ssl-hezwwgdphjgsdadtxckbgbxhjldw/Build/Intermediates.noindex/swift-nio-ssl.build/Debug-iphoneos/CNIOBoringSSL.build/DerivedSources/arm64 -I/Volumes/DerivedData-ramdisk/DerivedData/swift-nio-ssl-hezwwgdphjgsdadtxckbgbxhjldw/Build/Intermediates.noindex/swift-nio-ssl.build/Debug-iphoneos/CNIOBoringSSL.build/DerivedSources -F/Volumes/DerivedData-ramdisk/DerivedData/swift-nio-ssl-hezwwgdphjgsdadtxckbgbxhjldw/Build/Products/Debug-iphoneos -F/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/Frameworks -MMD -MT dependencies -MF /Volumes/DerivedData-ramdisk/DerivedData/swift-nio-ssl-hezwwgdphjgsdadtxckbgbxhjldw/Build/Intermediates.noindex/swift-nio-ssl.build/Debug-iphoneos/CNIOBoringSSL.build/Objects-normal/arm64/sha256-armv8.ios.aarch64.d --serialize-diagnostics /Volumes/DerivedData-ramdisk/DerivedData/swift-nio-ssl-hezwwgdphjgsdadtxckbgbxhjldw/Build/Intermediates.noindex/swift-nio-ssl.build/Debug-iphoneos/CNIOBoringSSL.build/Objects-normal/arm64/sha256-armv8.ios.aarch64.dia -c /Users/fabiansturm/Dropbox/projects/swift5.1/swift-nio-ssl/Sources/CNIOBoringSSL/crypto/fipsmodule/sha256-armv8.ios.aarch64.S -o /Volumes/DerivedData-ramdisk/DerivedData/swift-nio-ssl-hezwwgdphjgsdadtxckbgbxhjldw/Build/Intermediates.noindex/swift-nio-ssl.build/Debug-iphoneos/CNIOBoringSSL.build/Objects-normal/arm64/sha256-armv8.ios.aarch64.o

        /Users/fabiansturm/Dropbox/projects/swift5.1/swift-nio-ssl/Sources/CNIOBoringSSL/crypto/fipsmodule/sha256-armv8.ios.aarch64.S:75:10: error: expected compatible register or logical immediate
         tst w16,#ARMV8_SHA256
                 ^
        Command CompileC failed with a nonzero exit code

$ swift --version
Apple Swift version 5.1 (swiftlang-1100.0.266.1 clang-1100.0.32.1)
Target: x86_64-apple-darwin19.0.0

Operating system: macOS Catalina 10.15 Beta (19A526h)

$ uname -a
Darwin Mac1.local 19.0.0 Darwin Kernel Version 19.0.0: Tue Jul 23 01:19:36 PDT 2019; root:xnu-6153.0.103.151.1~4/RELEASE_X86_64 x86_64

### PR DESCRIPTION
**Pull request by fabstu**
Fixes #124 by making it compile again on said platforms.

@Lukasa is this the correct fix?

### CODE REVIEW

File: Tests/NIOSSLTests/NIOSSLIntegrationTest.swift [hunk 1]
```diff
@@ -21,7 +21,7 @@ import CNIOBoringSSL
 import NIO
 @testable import NIOSSL
 import NIOTLS
-import class Foundation.Process
+//import class Foundation.Process
```
Comment (Lukasa) [inline Tests/NIOSSLTests/NIOSSLIntegrationTest.swift]: I think commenting this out is unwise: if we don’t need it, we should delete it.

**Comment by fabstu** [comment]
Good to know I didn't realize. I removed my header-changes and then added it to `scripts/patch-2-arm-arch.patch` and regenerated the headers. I removed the commented line too.

**Comment by Lukasa** [comment]
I *think* it’s correct, but I’m away from my computer right now and need to double check. Does the branch now build on your iPad?

**Comment by fabstu** [comment]
Yes it does. Didn't expect it to be that easy to fix. Such small things can be quite annoying sometimes.

**Comment by Lukasa** [comment]
@swift-nio-bot Test this please

**Comment by Lukasa** [comment]
Great, shipped this fix in 2.3.1 which should unblock your issues. Thanks for contributing the patch! :sparkles:

**Pull request merged**